torch/_dynamo/variables/builder.py

```diff
@@ -1562,7 +1563,34 @@ def __eq__(self, other: object) -> bool:
 
 # Performs automatic dynamic dim determination.
 # Returns a CreateSymbolicPolicy
-def _automatic_dynamic(e, tx, name, static_shapes) -> CreateSymbolicPolicy:
+def _automatic_dynamic(
+    e, tx, name, static_shapes, recurse=True
+) -> CreateSymbolicPolicy:
+    if recurse and is_traceable_wrapper_subclass(e):
+        outer_policy = _automatic_dynamic(e, tx, name, static_shapes, recurse=False)
+
+        # Magic method allows a subclass to tune dynamic dim settings
+        # for inner tensors based on the outer policy
+        MARK_DYNAMIC_MAGIC_METHOD = "__tensor_mark_dynamic__"
+        if hasattr(e, MARK_DYNAMIC_MAGIC_METHOD):
+            getattr(e, MARK_DYNAMIC_MAGIC_METHOD)(outer_policy)
```

Comment:
I'm not that hot on the policy object making it to the public API; we've been refactoring it and there's a decent chance we'll keep refactoring it. My original thinking was that mark dynamic would only take a dim argument and be invoked when you called _dynamo.mark_dynamic.

This is not directly compatible with your needs, where you just want your thing to get unconditionally called even if there wasn't a user mark dynamic. But I think this is fine: you should just mark_dynamic your inner tensor (that is unconditionally called) in the constructor. Does this make sense?

Reply:
Ah okay, in that case I *will* bump the `__mark_dynamic__()` API to a possible future PR (if needed). My only use case wrt `NestedTensor` is the unconditional case you mentioned. Automatic dynamic works for me fine outside of this, and I don't think quantization needs the API either.